File: dev/20190531/session/api/fizzbuzz/src/main/java/fizzbuzz/domain/model/FizzBuzzList.java
package fizzbuzz.domain.model;

import fizzbuzz.application.IFizzBuzz;

import java.util.ArrayList;
import java.util.List;

public class FizzBuzzList implements IFizzBuzz {
  private final List<FizzBuzzValue> _list;

  public FizzBuzzList(List<FizzBuzzValue> list) {
    if (list.size() > 100)
    throw new IllegalArgumentException("FizzBuzzList can't generate over 100");
    _list = list;
  }

  @Override
  public String getValue() {
    return "";
  }

  public List<FizzBuzzValue> getList() {
    return _list;
  }

  public FizzBuzzList add(FizzBuzzValue value) {
    List<FizzBuzzValue> result = new ArrayList<>(_list);
    result.add(value);
    return new FizzBuzzList(result);
  }

}

File: dev/20190902/k2works/app/FizzBuzz/Domain/Model/FizzBuzzList.cs
using System;
using System.Collections.Generic;
using FizzBuzz.Application.Service;

namespace FizzBuzz.Domain.Model
{
    public class FizzBuzzList: IFizzBuzz
    {
        private readonly List<IFizzBuzz> _list;
        private static readonly int MAX_NUMBER = 100;

        public FizzBuzzList(List<IFizzBuzz> list)
        {
            if (list.Count > MAX_NUMBER) throw new ArgumentException("FizzBuzzList can't generate over 100''");
            _list = list;
        }

        public List<IFizzBuzz> List => _list;

        public FizzBuzzList Add(FizzBuzzValue value)
        {
            var result = new List<IFizzBuzz>(_list);
            result.Add(value);
            return new FizzBuzzList(result);
        }

        public string GetValue()
        {
            return "";
        }

        public List<IFizzBuzz> GetList()
        {
            return List;
        }
    }
}